### PR TITLE
bowerless build

### LINKED ISSUE #1573: remove bower in favor of NPM
- [ ] remove bower references in favor of NPM in build.gradle
- [ ] remove bower from doc
- [ ] remove bower checks from the apollo binary
- [ ] attempt to install webpack / dojo

### PR DESCRIPTION
**Pull request by nathandunn**
Merge dependent on ingest with PR:

https://github.com/GMOD/jbrowse/pull/906

Fixes #1573 

- [x] production version copy over
- ~~provide local ```package.json```  and have it point to client/apollo/js/main.js somehow?~~ (unnecessary)
- [x] make configurable from ```apollo-config.groovy```
- [x] make sure that it supports plugins

### CODE REVIEW

**Comment by nathandunn** [comment]
Fixes #1573

**Comment by nathandunn** [comment]
@deepakunni3  This is ready for review.   I would say try testing with your existing code.  If it doesn't work, try doing a `clean-all`

I tested via `run-local`, `deploy`, `run-app`.   I also tested adding plugins and custom versions, as well.

File: build.gradle [hunk 1]
```diff
@@ -297,54 +294,64 @@ task removeJBrowseDownload(type:Delete){
...
+        println "using npm to update: ${directory}"
         workingDir directory
-        commandLine "bower","update","-F","--allow-root"
+        commandLine "npm","update"
     }
 }
 
-def cloneRepoBower(String url,String directory,String branch){
+def cloneRepoNpm(String url, String directory, String branch){
+    println "init url ${url},${directory},${branch}"
```
Comment (deepakunni3) [inline build.gradle]: Remove additional `println`?

**Comment by deepakunni3** [comment]
@nathandunn run-local after a clean doesn't seem to download JBrowse and this breaks the build.

**Comment by nathandunn** [comment]
@deepakunni3 

K.  I will take a look.  Also, did you check your `apollo-config.groovy` in order to make sure you're not overriding it?  Prior versions of JBrowse will break it.

**Comment by deepakunni3** [comment]
I removed the config entirely. Relying on default configurations.

**Comment by nathandunn** [comment]
Thanks.  I'll take a look.

**Comment by nathandunn** [comment]
Did it work for the deploy, though?  If you haven’t tested it, no worries. 

Nathan

**Comment by deepakunni3** [comment]
@nathandunn Same story for run-local, deploy and release.

**Comment by nathandunn** [comment]
Thanks.  I'll take a look.

**Comment by nathandunn** [comment]
Can you copy and paste the output from one of those commands? 

Nathan

**Comment by nathandunn** [comment]
What version of npm and node are you using?

Nathan

**Comment by deepakunni3** [comment]
Here is what I get after `apollo run-local`:
```
Moving JBrowse into position gmod/jbrowse#f3292b59a1fa23558a08b38526e13ea0de9df941
:installJBrowse FAILED

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/deepak.unni3/GIT/apollo-nathandunn-too/build.gradle' line: 342

* What went wrong:
Execution failed for task ':installJBrowseNpm'.
> A problem occurred starting process 'command './setup.sh''

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED
```

and I am using npm 4.1.2

**Comment by nathandunn** [comment]
Can you try npm 5?  And can you add the browse output log? 

Nathan

**Comment by deepakunni3** [comment]
Tried with npm 5.3.0. Still the same issue.

Which browse output log?

**Comment by deepakunni3** [comment]
The problem is that JBrowse isn't being downloaded. And thus the error:
```
> A problem occurred starting process 'command './setup.sh''
```

Not sure whats causing it.

**Comment by nathandunn** [comment]
Ahh okay.  I’ve seen that before where it just hangs with the npm install. I have an idea how to fix this. Thanks.

**Comment by nathandunn** [comment]
works again with a clean download.  However, working on a stable ubuntu linux distro (16 I think) it fails:

```

ubuntu@ip-172-30-0-40:~/apollo-test-npm/target$ npm -v
3.5.2
ubuntu@ip-172-30-0-40:~/apollo-test-npm/target$ node -v
v4.2.6
```

It fails with this:

```
Moving JBrowse into position gmod/jbrowse#f3292b59a1fa23558a08b38526e13ea0de9df941
Legacy scripts wig-to-json.pl and bam-to-json.pl have removed from setup. Their functionality has been superseded by add-bam-track.pl and add-bw-track.pl. If you require the old versions, run 'setup.sh legacy'.
Installing javascript dependencies ...Dojo does not exist, installing
npm ERR! Linux 4.4.0-89-generic
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "install"
npm ERR! node v4.2.6
npm ERR! npm  v3.5.2

npm ERR! Cannot read property 'target' of null
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /home/ubuntu/apollo-test-npm/jbrowse-download/npm-debug.log
done
Gathering system information ...done
Installing Perl prerequisites ... done.

Formatting Volvox example data ... done.
To see the volvox example data, browse to http://your.jbrowse.root/index.html?data=sample_data/json/volvox.

Formatting Yeast example data ... done.
To see the yeast example data, browse to http://your.jbrowse.root/index.html?data=sample_data/json/yeast.
```

**Comment by nathandunn** [comment]
Using this: 

http://www.hostingadvice.com/how-to/install-nodejs-ubuntu-14-04/#node-version-manager  

FYI @enuggetry  having problems with a version of npm / node.  I'm going to run through a few things, but I'll let you know what I find.   Maybe during the setup it can catch the minimum npm / node versions.

**Comment by nathandunn** [comment]
- [ ] add minimum node requires node 5 (seems to be fine with npm 3)
- [ ] look at npm hanging sometimes and recommendations

**Comment by nathandunn** [comment]
just updated to add newest jbrowse, but no fundamendatl changes

**Comment by nathandunn** [comment]
@deepakunni3  ready to review

**Comment by deepakunni3** [comment]
@nathandunn Tested and works for `run local` and `deploy`. 👍

**Pull request merged**